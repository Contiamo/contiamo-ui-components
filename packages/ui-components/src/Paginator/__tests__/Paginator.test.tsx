import * as React from "react"
import { render, mount } from "enzyme"

import ThemelessPaginator from "../Paginator"
import wrapDefaultTheme from "../../../utils/wrap-default-theme"

const Paginator = wrapDefaultTheme(ThemelessPaginator)

class StatefulPaginator extends React.Component {
  state = {
    page: 2
  }

  handleChange (newPage) {
    this.setState(prevState => ({page: newPage}))
  }

  render() {
    return (
      <Paginator
        pageCount={5}
        selected={this.state.page}
        onChange={newPage => { this.handleChange(newPage) }}
      />
    )
  }
}

describe("Paginator Component", () => {
  it("Should initialize properly", () => {
    expect(render(<Paginator pageCount={10} onChange={newPage => {}} />)).toMatchSnapshot()
  })
  it("Should respond to page changes", () => {
    const testFn = jest.fn()
    const comp = mount(<Paginator pageCount={5} onChange={testFn} />)
    comp.find('li').last().simulate('click')
    expect(testFn).toHaveBeenCalled()
  })
  it('Should update state of container component', () => {
    const comp = mount(<StatefulPaginator />)
    comp.find('li').at(3).simulate('click')
    expect(comp.state().page).toBe(3)
  })
})