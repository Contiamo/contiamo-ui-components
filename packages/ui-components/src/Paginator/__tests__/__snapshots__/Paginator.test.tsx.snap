// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component Should initialize properly 1`] = `
<div
  class="css-1ef2t77"
>
  <div
    class="css-1f5s5d"
    style="display:flex;user-select:none"
  >
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      <svg
        height="11"
        viewbox="0 0 24 24"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          fill="none"
          points="11 17 6 12 11 7"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <polyline
          fill="none"
          points="18 17 13 12 18 7"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      <svg
        fill="none"
        height="11"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewbox="0 0 24 24"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="15 18 9 12 15 6"
        />
      </svg>
    </div>
    <div
      class="css-13al2u6"
      role="button"
      tabindex="-1"
    >
      1
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      2
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      3
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      4
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      5
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      ...
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      10
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      <svg
        fill="none"
        height="11"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewbox="0 0 24 24"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="9 18 15 12 9 6"
        />
      </svg>
    </div>
    <div
      class="css-13uzrop"
      role="button"
      tabindex="-1"
    >
      <svg
        height="11"
        viewbox="0 0 24 24"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          fill="none"
          points="13 17 18 12 13 7"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <polyline
          fill="none"
          points="6 17 11 12 6 7"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
